client/engine/engine.go

```diff
@@ -167,9 +167,14 @@ func (e *Engine) handleMessage(message protocols.Message) (ObjectiveChangeEvent,
 		if err != nil {
 			return ObjectiveChangeEvent{}, err
 		}
-
 		allCompleted.CompletedObjectives = append(allCompleted.CompletedObjectives, progressEvent.CompletedObjectives...)
 
+		relatedProgressEvent, err := e.attemptProgressForRelatedObjectives(&updatedObjective)
```

Comment:
Maybe:

```suggestion
		relatedObjectiveCompletions, err := e.attemptProgressForRelatedObjectives(&updatedObjective)
```
?

Out-of-scope, but ObjectiveChangeEvent is probably too broad a name for what it currently is: a container storing lists of completed objectives.